core/system_metric/sys_metric_stat.go

```diff
@@ -156,16 +263,95 @@ func InitCpuCollector(intervalMs uint32) {
 }
 
 func retrieveAndUpdateCpuStat() {
-	cpuPercent, err := getProcessCpuStat()
-	if err != nil {
-		logging.Error(err, "Fail to retrieve and update cpu statistic")
-		return
+	var (
+		cpuPercent float64
+		err        error
+	)
+	if isContainer {
+		cpuPercent, err = GetContainerCpuStat()
+		if err != nil {
+			logging.Error(err, "Fail to retrieve and update cpu statistic")
+			return
+		}
+	} else {
+		cpuPercent, err = getProcessCpuStat()
+		if err != nil {
+			logging.Error(err, "Fail to retrieve and update cpu statistic")
+			return
+		}
 	}
+
 	metrics.SetCPURatio(cpuPercent)
 	currentCpuUsage.Store(cpuPercent)
 }
 
-// getProcessCpuStat gets current process's memory usage in Bytes
+func GetContainerCpuStat() (float64, error) {
+
+	var (
+		currentSysCpuTotal       float64
+		currentContainerCpuTotal float64
+		err                      error
+	)
+
+	currentSysCpuTotal, err = getSysCpuUsage()
+	if err != nil {
+		return 0, err
+	}
+	currentContainerCpuTotal, err = getContainerCpuUsage()
+	if err != nil {
+		return 0, err
+	}
+
+	preSysTotalCpu, ok := preSysTotalCpuUsage.Load().(float64)
+	if !ok {
+		return 0, errors.New("preSysTotalCpuUsage load is not float64")
+	}
+	preContainerCpu, ok := preContainerCpuUsage.Load().(float64)
+
+	if !ok {
+		return 0, errors.New("preContainerCpuUsage load is not float64")
+	}
+	if currentSysCpuTotal-preSysTotalCpu == 0 {
+		return 0, err
```

Comment:
`return 0, nil`?

Reply:
the method of calculating cpu is similar to docker stats ,refer this 
`https://stackoverflow.com/questions/45196440/cpupercent-metric-from-docker-stats-vs-cgroups`